src/registry/routes/helpers/get-component-fallback.js

```diff
@@ -4,28 +4,63 @@ var request = require('minimal-request');
 var url = require('url');
 var _ = require('underscore');
 
-module.exports = function(fallbackRegistryUrl, headers, component, callback) {
-  return request({
-    method: 'post',
-    url: fallbackRegistryUrl,
-    headers: _.extend({}, headers, {'host': url.parse(fallbackRegistryUrl).host}),
-    json: true,
-    body: {
-      components: [
-        component
-      ]
+module.exports = {
+  getComponent: function (fallbackRegistryUrl, headers, component, callback) {
+    return request({
+      method: 'post',
+      url: fallbackRegistryUrl,
+      headers: _.extend({}, headers, {'host': url.parse(fallbackRegistryUrl).host}),
+      json: true,
+      body: {
+        components: [
+          component
+        ]
+      }
+    }, function (err, res) {
+      if (err || !res || res.length === 0) {
+        return callback({
+          status: 404,
+          response: {
+            code: 'NOT_FOUND',
+            error: err
+          }
+        });
+      }
+
+      return callback(res[0]);
+    });
+  },
+  getComponentInfoOrPreviewFallback: function (conf, req, res, localRegistryError, component, callback) {
+    if (!localRegistryError) {
+      return callback(null, component);
     }
-  }, function(err, res) {
-    if(err || !res || res.length === 0) {
-      return callback({
-        status: 404,
-        response: {
-          code: 'NOT_FOUND',
-          error: err
-        }
-      });
+
+    if (!conf.fallbackRegistryUrl) {
+      return callback({localError: localRegistryError});
     }
 
-    return callback(res[0]);
-  });
-};
\ No newline at end of file
+    var path = req.originalUrl;
+    return request({
+      method: 'get',
+      url: conf.fallbackRegistryUrl + path.substr(1, path.length - 1),
```

Comment:
`req.originalUrl` comes with a `/` as first character and we already have trailing `/` in `conf.fallbackRegistryUrl` is it prefered if I used `url` library or is this fine?